### PR TITLE
[UM.5.5.r1] MSM8939, MSM8936 SoC Porting + SoMC Tulip

### PR DESCRIPTION
**Pull request by kholk**
This is the complete MSM8939, MSM8936 (and 8916) SoC porting patchset, including SoMC Kanuti platform, Tulip board.

Note: Make sure it's not breaking other devices build prior merging.
Note 2: There is a cute Adreno patch, boot loire and tone prior doing anything 
Note 3: That's old stuff, so new commits from 3.10 kernel must be ported
Note 7: On fire, I think................ :°)

### CODE REVIEW

**Comment by bartcubbins** [comment]
@kholk test this?:)

**Comment by jerpelea** [comment]
that is the ideea but please note that patches from 3.10 and binaries from dw

**Comment by kholk** [comment]
@bartcubbins probably requires patches from 3.10 to run correctly

**Comment by bartcubbins** [comment]
@kholk ok I'll port all requires patches. thank you

File: drivers/clk/msm/clock-cpu-8936.c [hunk 1]
```diff
@@ -170,8 +310,8 @@ static void get_speed_bin(struct platform_device *pdev, int *bin,
 
 	*bin = (pte_efuse >> 2) & 0x7;
 
-	dev_info(&pdev->dev, "Speed bin: %d PVS Version: %d\n", *bin,
-								*version);
+out:
```
Comment (jerpelea) [inline drivers/clk/msm/clock-cpu-8936.c]: drivers/clk/msm/clock-cpu-8936.c: In function 'get_speed_bin':
drivers/clk/msm/clock-cpu-8936.c:313:1: warning: label 'out' defined but not used [-Wunused-label]

**Review by jerpelea — COMMENTED** [review]
fix it

**Comment by jerpelea** [comment]
In file included from ./arch/arm64/include/asm/page.h:49:0,
                 from include/linux/mmzone.h:20,
                 from include/linux/gfp.h:5,
                 from include/linux/mm.h:9,
                 from include/linux/mman.h:4,
                 from arch/arm64/mm/mmu.c:24:
arch/arm64/mm/mmu.c: In function 'mem_text_address_writeable':
arch/arm64/mm/mmu.c:134:31: error: 'prot_sect_kernel' undeclared (first use in this function)
    __pmd(__pa(addr_aligned) | prot_sect_kernel));
                               ^
./arch/arm64/include/asm/pgtable-types.h:70:19: note: in definition of macro '__pmd'
 #define __pmd(x) (x)
                   ^
arch/arm64/mm/mmu.c:134:31: note: each undeclared identifier is reported only once for each function it appears in
    __pmd(__pa(addr_aligned) | prot_sect_kernel));
                               ^
./arch/arm64/include/asm/pgtable-types.h:70:19: note: in definition of macro '__pmd'
 #define __pmd(x) (x)
                   ^

**Comment by pablomh** [comment]
- Please remove: arch/arm/boot/dts/qcom/msm8939-kanuti_tulip_common.dtsi.rej
- Did any 8936 CPU ever shipped in a Sony product? Is there any point to keep those DT around?

**Comment by bartcubbins** [comment]
@jerpelea @kholk I have a fix for this issues but the kernel is still not booting on tulip

**Comment by jerpelea** [comment]
you need to port the last 3 PR from 3.10

**Comment by pablomh** [comment]
Any way we can test this on tulip? I'm using SonyAosp android-7.1 branch but apart from missing blobs I'm having build problems, so my testing is reduced to a kernel compilation testing...

**Comment by kholk** [comment]
You should first make sure that the kernel compiles fine..... then, about the blobs thingy, for a raw first android boot, they are not REALLY necessary. You can boot with your 3.10 1.3.3 blobs, it will work fine (I did that multiple times).

Of course, fine means you can see the lockscreen, launcher and settings menu just fine, nothing fancy.

**Comment by bartcubbins** [comment]
@kholk I fix all errors and I can compile the kernel but the tulip still not boot

**Comment by kholk** [comment]
Check the pstore logs !

**Comment by bartcubbins** [comment]
@kholk I can't get logs. I see Sony logo 2-3 sec then reboot

**Comment by pablomh** [comment]
Regarding the error in mmu.c:

```
arch/arm64/mm/mmu.c: In function 'mem_text_address_writeable':
arch/arm64/mm/mmu.c:134:31: error: 'prot_sect_kernel' undeclared (first use in this function)
__pmd(__pa(addr_aligned) | prot_sect_kernel));
```

I think that the error was introduced in the following commits:

https://github.com/sonyxperiadev/kernel/commit/f15f3173d568313fa6916eebd967d1436b4c13e1
https://github.com/sonyxperiadev/kernel/commit/7960233d590fbc5d4012388e2f2fbddadedb4c2f

and it's hidden behind an #ifdef CONFIG_STRICT_MEMORY_RWX.

**Comment by bartcubbins** [comment]
@kholk I got logs: https://hastebin.com/lotupikuhi.go

**Comment by pablomh** [comment]
Please take a look at https://github.com/kholk/kernel/pull/8. With it, the kernel builds successfully for tulip :)

**Comment by kholk** [comment]
PR merged on my GIT. Anyway, the logs by @bartcubbins are showing both a MDSS problem AND a kernel configuration issue.

**Comment by bartcubbins** [comment]
@kholk I added some missing configs from 1.3.3 kernel and I got these logs: https://hastebin.com/avumuwujih.vbs

**Comment by pablomh** [comment]
@bartcubbins, would you mind PR'ing those missing config changes so everyone is in the same page? Thanks!!

**Pull request closed without merging**